### PR TITLE
Fix text-break in IE and Edge Legacy

### LINKED ISSUE #29319: Docs - Word break
I was trying to apply the `text-break` class in order to wrap text. Everything was working fine in Chrome. But once I moved over to IE and Edge, things were all over the place. I then visited the docs site and saw that even the docs site doesn't exhibit the intended behavior.

image

**Mandatory Information:**
Windows 10 Version 1709 / 16299.1331
Internet Explorer v. 11.1331.16299.0 / 11.0.140 (KB4511872)
Microsoft Edge 41.16299.1004.0 / Microsoft EdgeHTML 16.16299

### PR DESCRIPTION
**Pull request by ihorzenich**
Fix #29319 again.
Correcting Backport #30932 to v4

`word-break` didn't support `break-word` value in IE and Edge Legacy 
`overflow-wrap` is absent in IE/Edge Legacy (it's modern replace for `word-wrap`)
So we need to add back `word-wrap: break-word` for IE and Edge Legacy.

### CODE REVIEW

**Comment by XhmikosR** [comment]
Thanks for the PR, but please always make sure CI passes.

**Comment by ihorzenich** [comment]
@XhmikosR thanks, could you please clarify how should I proceed with Lint errors in files below?
```
/home/runner/work/bootstrap/bootstrap/js/src/button.js
/home/runner/work/bootstrap/bootstrap/js/src/modal.js
/home/runner/work/bootstrap/bootstrap/js/src/scrollspy.js
```
I didn't touch them, they're the same as current upstream v4-dev branch.

**Comment by XhmikosR** [comment]
Those are not errors, they are warnings and unrelated. https://github.com/twbs/bootstrap/pull/31727/checks?check_run_id=1152664809#step:6:53

**Comment by XhmikosR** [comment]
BTW I see why you got confused; GitHub changed something in regard to Actions and the Stylelint errors are not highlighted like the warnings or the parent failed commit. Either way it should be pretty trivial to fix, but for future reference better clone the repo locally so that you run the tests.

**Comment by ihorzenich** [comment]
@XhmikosR thanks, I've fixed lint error and updated PR, please review.

**Comment by XhmikosR** [comment]
@twbs/css-review can I have a review just so that we are safe please?

**Comment by ihorzenich** [comment]
@XhmikosR I've added more detailed comments how .text-break rules works.

File: scss/utilities/_text.scss [hunk 1]
```diff
@@ -63,8 +63,19 @@
 .text-decoration-none { text-decoration: none !important; }
 
 .text-break {
-  word-break: break-word !important; // IE & < Edge 18
-  overflow-wrap: break-word !important;
+  // We wan't to use `overflow-wrap: anywhere` to avoid issues with flex containers
```
Comment (XhmikosR) [inline scss/utilities/_text.scss:66]: @mdo this whole section needs grammar fixes and personally I find it too verbose

**Comment by mdo** [comment]
See #31793.

**Pull request closed without merging**